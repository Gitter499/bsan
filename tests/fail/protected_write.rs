//@run: 1
fn main() {
    let mut x = 1;
    let rx = &mut x;
    let ptr_x = rx as *mut i32;
    foo(rx, ptr_x);
}

fn foo(_r: &mut i32, p: *mut i32) {
    unsafe { *p = 1 };
}