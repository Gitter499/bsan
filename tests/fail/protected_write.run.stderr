An error occurred: UndefinedBehavior(AliasingViolation(TreeError { error_kind: ProtectedDisabled(Permission { inner: ReservedFrz { conflicted: false } }), alloc_id: ALLOC, error_offset: 0, conflicting_info: NodeDebugInfo { tag: <TAG>, name: None, history: History { tag: <TAG>, created: (Span(0), Permission { inner: ReservedFrz { conflicted: false } }), events: [] } }, access_cause: Explicit(Write), accessed_info: NodeDebugInfo { tag: <TAG>, name: None, history: History { tag: <TAG>, created: (Span(0), Permission { inner: ReservedFrz { conflicted: false } }), events: [Event { transition: PermTransition { from: ReservedFrz { conflicted: false }, to: Active }, access_cause: Explicit(Write), is_foreign: false, access_range: Some(AllocRange { start: size0, size: size4 }), transition_range: 0..4, span: Span(0) }] } } }))


